### ISSUE TITLE
Focusing a radio button and validation on focus

### ISSUE THREAD

**Issue by MiMaMuh** [davidkpiano/react-redux-form#428]
Hi, when I ```focus``` a radio-button with a keyboard tab and do not check one of the radios, the ```validators``` get the ```value``` of inputs' value-attribute passed instead of the stores' model value:
```jsx
<Field
    model={'model.test'}
    validateOn={['change', 'focus']}
    validators={{
        exampleValidator: (value) => {
            console.log(value); // value === 'test_1' on focus even if value of the model === null
        },
    }}
>
    <input type="radio" value="test_1" name="group" />
</Field>
```
In this example, when I ```focus``` the radio button and the model value of ```model.test```is currently null, the validator gets the value of the radio-buttons' value-attribute passed, in this example ```test_1``` when the first radio-button is focused.

**Comment by davidkpiano** [comment]
Hm, I'm confused - the behavior seems correct. Validators are based on the value that the model is about to become rather than what the model is previously, which would be the value toggled by the radio button. Can you clarify the use-case?

**Comment by MiMaMuh** [comment]
Yes, of course. I think the issue is, that the model will not change when I just tab over the input without selecting it. But I do get the value-attribute-value of the input passed to the validator with isn't correct when nothing is checked I think. 

**There are two use-cases:**

**I. I have a group of two radio-buttons** where I have to select one of the two options but none is selected currently:
```jsx
<Field
   ...
>
    <input type="radio" value="male" name="gender" />
    <input type="radio" value="female" name="gender" />
</Field>
```
Now a user uses its keyboard and jumps from a previous input to my radio-button-group. In this case, none of the radio-buttons is selected till the user clicks on one of the radio-buttons or press spacebar on its keyboard. For some reasons the user do not select one of the radios but just tab again to jump to the next input. 

The model of gender is still ```null```but the ```onBlur``` validator gets the radio-box-value passed which was focused: ```male``` for example. The input should show an error, because the user has to select one of the options but nothing happens, as the validator says: Hey, I got ```male```but the model says: You are a liar, I got nothing. :)

**II. I have a checkbox for legal reasons** to confirm the privacy policy or similar:
<Field
   ...
>
    <input type="checkbox" value="privacy_policy" />
</Field>
The same scenario: User uses its keyboard and jumps over the checkbox without checking it. The ```onBlur``` validator should check if it was checked because it is required otherwise show an error. The model is still ```null``` but the validator gets ```privacy_policy``` as its value ...

I hope I could clarify it?
